packages/backend/src/logger/customLogger.ts

```diff
@@ -42,15 +50,38 @@ const transports = {
       ),
     }),
   ],
-  auditLog: [
-    new winston.transports.Console({
-      format: winston.format.combine(
-        auditLogFormat({ isAuditLog: true }),
-        defaultFormat,
-        winston.format.json(),
-      ),
-    }),
-  ],
+  auditLog: (config?: Config) => {
+    if (config?.getOptionalBoolean('console.enabled') === false) {
+      return [];
+    }
+    return [
+      new winston.transports.Console({
+        format: auditLogWinstonFormat,
+      }),
+    ];
+  },
+  auditLogFile: (config?: Config) => {
+    if (!config?.getOptionalBoolean('rotateFile.enabled')) {
+      return [];
+    }
+    return [
+      new winston.transports.DailyRotateFile({
+        format: auditLogWinstonFormat,
+        dirname:
+          config?.getOptionalString('rotateFile.logFileDirPath') ??
+          '/var/log/redhat-developer-hub/audit',
+        filename:
+          config?.getOptionalString('rotateFile.logFileName') ??
+          'redhat-developer-hub-audit-%DATE%.log',
+        datePattern: config?.getOptionalString('rotateFile.dateFormat'),
+        frequency: config?.getOptionalString('rotateFile.frequency'),
+        zippedArchive: config?.getOptionalBoolean('rotateFile.zippedArchive'),
+        utc: config?.getOptionalBoolean('rotateFile.utc'),
+        maxSize: config?.getOptionalString('rotateFile.maxSize'),
+        maxFiles: config?.getOptional('rotateFile.maxFilesOrDays'),
+      }),
+    ];
+  },
```

Comment:
Thinking about this more, maybe we should make this one function that appends the selected transport to an array. Thoughts?